tests/db_functions/tests.py

```diff
@@ -547,3 +547,74 @@ def test_now(self):
             ['How to Time Travel'],
             lambda a: a.title
         )
+
+    def test_length_transform(self):
+        try:
+            CharField.register_transform(Length, 'length')
```

Comment:
We added the ability to use `register_lookup()` as a context manager to avoid `try/finally` (see 13dca01af02e91f25ae1d96b09b857993714c96d). Could you do that here?